model/common/src/icon4py/model/common/decomposition/definitions.py

```diff
@@ -69,6 +70,7 @@ class EntryType(IntEnum):
     @builder.builder
     def with_dimension(self, dim: Dimension, global_index: np.ndarray, owner_mask: np.ndarray):
         masked_global_index = ma.array(global_index, mask=owner_mask)
+        masked_global_index = xp.asarray(masked_global_index)
         self._global_index[dim] = masked_global_index
```

Comment:
I guess that won't work. I don't now what happens with the mask if you do that and we need it that mask later. But indeed cupy does not implement the numpy's masked_array interface. We would have to do a simple hand made implementation which should be enough for what we need.   I'll open a fix for that.